woocommerce/woocommerce-gateway-stripe
# Flag when the account is connected via App and if keys have expired

Fixes #3349 

## Changes proposed in this Pull Request:

In Stripe 8.6.0 we added support for merchants to connect to the WooCommerce Stripe App if they have account keys and those keys are for an account that is already linked to a different App.

Stores that are connected via the WooCommerce Stripe App have expiring keys and so we understanding if the store is connected via the App is important. 

This PR adds a couple of UI elements to better communicate the connection type and if the keys have expired.

| Normal connection | Connected via App |
|--------|--------|
| <img width="624" alt="Screenshot 2024-08-12 at 3 39 37 PM" src="https://github.com/user-attachments/assets/274aad6a-5e0a-4c14-b03c-fa3c3c1d1732"> | <img width="634" alt="Screenshot 2024-08-12 at 2 27 12 PM" src="https://github.com/user-attachments/assets/bd0979be-ef04-4e1c-b231-7ee39dd45f39"> | 

If the account keys have expired, it will show an expired status.
<p align="center">
<img width="400" alt="Screenshot 2024-08-12 at 3 37 16 PM" src="https://github.com/user-attachments/assets/13f87baa-4d07-41db-b7ce-378c3193ad0a">
</p> 



## Testing instructions

1. Set your Stripe API keys to an account that is connected to a different App. 
2. Go to your Stripe plugin settings.
3. Open the Account connection modal.
4. Hover the **Create or connect to a test account** button.
5. The URL should say `marketplace.stripe.com/...`.
6. Click the button.
7. Connect to an account. (it doesn't have to be the same account)
8. Complete the flow to connect to the App. 
9. After returning to the store on this branch you will see the `Connected 🔗` status (see screenshot above). 
10. Go to **Tools > Scheduled Actions** and search for the **`wc_stripe_refresh_connection`** hook. 
11. Cancel the `wc_stripe_refresh_connection` action. 
    - We want the keys to expire.
12. Wait 1 hour+ (I know sorry). I don't know if there's a way to speed this up. 
13. Once you're keys have expired, refresh the settings and you should see the **Keys expired ⛓️‍💥** account status (see screenshot above). 
14. Click the **Create or connect an account** button again. 
15. You should be on the standard connect flow now. 
16. Connect to an account as normal.
17. Once you return to the site, you should see the **Connected** status with no link tooltip. 




---


**Post merge**
